_rules/inline-link-in-paragraph-has-different-background-color-and-distinguishable-style-on-hover-and-focus-2803b8.md

```diff
@@ -0,0 +1,143 @@
+---
+id: 2803b8
+name: Inline link has different background color and distinguishable style on hover and focus
+rule_type: atomic
+description: |
+  This rule checks that inline links with different background color have a style on hover and focus that distinguishes them from the surrounding text
+accessibility_requirements:
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], for which all the following is true:
+
+- **link text**: The element has [visible][] [text nodes][text node] as [descendants][descendant] in the [flat tree][]; and
+- **non-link line text**: The element is [rendered on a line][] containing [visible][] [text nodes][text node] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][].
+
+## Expectation
+
+Each target element has:
+
+- a [highest possible contrast][] that is at least 3.0:1 between its [background color][] and the [background color][] of the [visible][] [text nodes][text node] [rendered on the same line][rendered on a line] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][]; and
+- a [distinguishing style][] both when the target element [gains focus][focused] and the target element is [hovered][].
+
+## Assumptions
+
+- The link is distinguishable from the rest of the text with color, which means it fails SC 1.4.1 when there is not another way to distinguish it.
+- The 3:1 contrast difference between text is minimal to what would be sufficient to meet WCAG 2.0. This value is part of [technique G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183), but is not specified in the [1.4.1 success criterion](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html).
+- Any change in font is sufficiently distinguishable, and fonts are loaded when they are present.
+- If multiple colors are used in the visible text nodes of the line then color can not be a distinguishing factor.
```

Comment:
@WilcoFiers The reason for it is, if you have different background colors in the different non-link text nodes then you can't use the background color to distinguish the link node from the other nodes (since there will be multiple colors at play, not just 2).

I've rephrased the assumption. Hopefully it is now easier to understand.